sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,247 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImport(models.TransientModel):
+    _inherit = "sale.order.import"
+
+    import_type = fields.Selection(
+        selection_add=[("edifact", "EDIFACT")], ondelete={"edifact": "cascade"}
+    )
+    # Make doc_type field support for EDIFACT type
+    doc_type = fields.Selection(readonly=False)
+    parsed_json = fields.Text()
```

Comment:
```suggestion
    json_data_preview = fields.Text(help="Used by the btn `button_parse_order_preview` to preview data before importing")
```
And I would add a TODO to move this feature to the base module or to an additional module.